### PR TITLE
COLLECTIONS-777 JUnit v5 assertThrows no2

### PR DESCRIPTION
**Pull request by nhojpatrick**
Another set of changed, should not have merge conflicts with other PR.

### CODE REVIEW

**Comment by nhojpatrick** [comment]
Something has gone wrong, will fix later

**Comment by nhojpatrick** [comment]
Same type of work as #281 covered by COLLECTIONS-777 and have also squashed and prefixed.

File: src/test/java/org/apache/commons/collections4/comparators/ComparatorChainTest.java [hunk 1]
```diff
@@ -67,11 +68,8 @@ public void testBadNoopComparatorChain() {
         final ComparatorChain<Integer> chain = new ComparatorChain<>();
         final Integer i1 = 4;
         final Integer i2 = 6;
-        try {
-            chain.compare(i1, i2);
-            fail("An exception should be thrown when a chain contains zero comparators.");
-        } catch (final UnsupportedOperationException e) {
-        }
+
+        assertThrows(UnsupportedOperationException.class, () -> chain.compare(i1, i2));
```
Comment (kinow) [inline src/test/java/org/apache/commons/collections4/comparators/ComparatorChainTest.java:72]: I think the error message would be helpful in this case :point_up: maybe add that to the `assertThrows`

File: src/test/java/org/apache/commons/collections4/iterators/ArrayListIteratorTest.java [hunk 2]
```diff
@@ -104,15 +105,8 @@ public void testListIteratorSet() {
...
-            iter.set((E) "should fail");
-            fail("ListIterator#set should fail if next() or previous() have not yet been called.");
-        } catch (final IllegalStateException e) {
-            // expected
-        } catch (final Throwable t) { // should never happen
-            fail(t.toString());
-        }
-
+        ListIterator<E> finalIter = iter;
+        assertThrows(IllegalStateException.class, () -> finalIter.set((E) "should fail"));
```
Comment (kinow) [inline src/test/java/org/apache/commons/collections4/iterators/ArrayListIteratorTest.java:109]: Would be good to preserve this error message too, IMO.

File: src/test/java/org/apache/commons/collections4/iterators/ListIteratorWrapper2Test.java [hunk 3]
```diff
@@ -109,11 +110,7 @@ public void testRemove() {
         assertEquals(-1, iter.previousIndex());
         assertEquals(0, iter.nextIndex());
 
-        try {
-            iter.remove();
-            fail("ListIteratorWrapper#remove() should fail; must be initially positioned first");
-        } catch (final IllegalStateException e) {
-        }
+        assertThrows(IllegalStateException.class, () -> iter.remove());
```
Comment (kinow) [inline src/test/java/org/apache/commons/collections4/iterators/ListIteratorWrapper2Test.java:113]: Ditto here...

File: src/test/java/org/apache/commons/collections4/iterators/ListIteratorWrapper2Test.java [hunk 4]
```diff
@@ -134,11 +131,7 @@ public void testRemove() {
         assertEquals(-1, iter.previousIndex());
         assertEquals(0, iter.nextIndex());
 
-        try {
-            iter.remove();
-            fail("ListIteratorWrapper#remove() should fail; must be repositioned first");
-        } catch (final IllegalStateException e) {
-        }
+        assertThrows(IllegalStateException.class, () -> iter.remove());
```
Comment (kinow) [inline src/test/java/org/apache/commons/collections4/iterators/ListIteratorWrapper2Test.java:134]: And here...

File: src/test/java/org/apache/commons/collections4/iterators/ObjectArrayListIteratorTest.java [hunk 5]
```diff
@@ -101,15 +102,8 @@ public void testListIteratorSet() {
...
-            iter.set((E) "should fail");
-            fail("ListIterator#set should fail if next() or previous() have not yet been called.");
-        } catch (final IllegalStateException e) {
-            // expected
-        } catch (final Throwable t) { // should never happen
-            fail(t.toString());
-        }
-
+        ListIterator<E> finalIter = iter;
+        assertThrows(IllegalStateException.class, () -> finalIter.set((E) "should fail"));
```
Comment (kinow) [inline src/test/java/org/apache/commons/collections4/iterators/ObjectArrayListIteratorTest.java:106]: Ditto.

File: src/test/java/org/apache/commons/collections4/collection/PredicatedCollectionTest.java [hunk 6]
```diff
@@ -99,12 +98,9 @@ public void testIllegalAddAll() {
...
         elements.add((E) Integer.valueOf(3));
         elements.add((E) "four");
-        try {
-            c.addAll(elements);
-            fail("Integer should fail string predicate.");
-        } catch (final IllegalArgumentException e) {
-            // expected
-        }
+
+        assertThrows(IllegalArgumentException.class, () -> c.addAll(elements));
```
Comment (kinow) [inline src/test/java/org/apache/commons/collections4/collection/PredicatedCollectionTest.java:102]: Maybe keep the `fail` message here.

File: src/test/java/org/apache/commons/collections4/collection/PredicatedCollectionTest.java [hunk 7]
```diff
@@ -82,12 +84,9 @@ public PredicatedCollectionTest(final String name) {
...
         final Collection<E> c = makeTestCollection();
         final Integer i = 3;
-        try {
-            c.add((E) i);
-            fail("Integer should fail string predicate.");
-        } catch (final IllegalArgumentException e) {
-            // expected
-        }
+
+        assertThrows(IllegalArgumentException.class, () -> c.add((E) i));
```
Comment (kinow) [inline src/test/java/org/apache/commons/collections4/collection/PredicatedCollectionTest.java:88]: Maybe keep the `fail` message here.

**Review by kinow — COMMENTED** [review]
Great job @nhojpatrick ! I think some of the `fail` messages could be useful, so keeping them around would be better. I think we can include it in the `assertThrows` as a parameter. The rest looks good to me! Thanks

**Comment by nhojpatrick** [comment]
@kinow @garydgregory could you both confirm what style you require? each pr I feel I need to do a different style, one PR I get told to keep comments and check thrown exception message and next PR i'm being asked to not do that due to maintainer overhead. I'm happy to do either way.

I've got 673 more assertions on my list to refactor, so would like to know beforehand what I aiming for.

File: src/test/java/org/apache/commons/collections4/collection/PredicatedCollectionTest.java (hunk 7 above)
Comment (nhojpatrick) [reply src/test/java/org/apache/commons/collections4/collection/PredicatedCollectionTest.java:88]: added back in next push.

File: src/test/java/org/apache/commons/collections4/collection/PredicatedCollectionTest.java (hunk 6 above)
Comment (nhojpatrick) [reply src/test/java/org/apache/commons/collections4/collection/PredicatedCollectionTest.java:102]: ditto too

File: src/test/java/org/apache/commons/collections4/comparators/ComparatorChainTest.java (hunk 1 above)
Comment (nhojpatrick) [reply src/test/java/org/apache/commons/collections4/comparators/ComparatorChainTest.java:72]: ditto too

File: src/test/java/org/apache/commons/collections4/iterators/ArrayListIteratorTest.java (hunk 2 above)
Comment (nhojpatrick) [reply src/test/java/org/apache/commons/collections4/iterators/ArrayListIteratorTest.java:109]: ditto

File: src/test/java/org/apache/commons/collections4/iterators/ListIteratorWrapper2Test.java (hunk 3 above)
Comment (nhojpatrick) [reply src/test/java/org/apache/commons/collections4/iterators/ListIteratorWrapper2Test.java:113]: ditto too

File: src/test/java/org/apache/commons/collections4/iterators/ListIteratorWrapper2Test.java (hunk 4 above)
Comment (nhojpatrick) [reply src/test/java/org/apache/commons/collections4/iterators/ListIteratorWrapper2Test.java:134]: ditto

File: src/test/java/org/apache/commons/collections4/iterators/ObjectArrayListIteratorTest.java (hunk 5 above)
Comment (nhojpatrick) [reply src/test/java/org/apache/commons/collections4/iterators/ObjectArrayListIteratorTest.java:106]: ditto

**Comment by garydgregory** [comment]
@nhojpatrick , @kinow ,
@ me in a message when this PR is ready for a final review.

**Comment by nhojpatrick** [comment]
@garydgregory ready for final review, each pr is different files.

**Pull request merged**